fvt/storage/storage_test.go

```diff
@@ -0,0 +1,189 @@
+var isvcWithPvcInStorageConfig = []string{"isvc-pvc-storage-uri", "isvc-pvc-storage-path", "isvc-pvc2"}
+
+// ISVC using PVC not in the FVT `storage-config` Secret (config/dependencies/fvt.yaml)
+// this should work only after setting allowAnyPVC = true
+var isvcWithPvcNotInStorageConfig = "isvc-pvc3"
+
+// ISVC using a PVC that does not exist at all, this ISVC should fail to load
+var isvcWithNonExistentPvc = "isvc-pvc4"
+
+var _ = Describe("ISVCs", Ordered, FlakeAttempts(3), func() {
+
+	Describe("with PVC in storage-config", Ordered, func() {
+
+		for _, name := range isvcWithPvcInStorageConfig {
+
+			Describe("\""+name+"\"", Ordered, func() {
+				var isvcName = name
+				var fileName = isvcFiles[name]
+
+				It("should successfully load a model", func() {
+					isvcObject := NewIsvcForFVT(fileName)
+					isvcName = isvcObject.GetName()
+					CreateIsvcAndWaitAndExpectReady(isvcObject, PredictorTimeout)
+				})
+
+				It("should successfully run inference", func() {
+					ExpectSuccessfulInference_sklearnMnistSvm(isvcName)
+				})
+
+				BeforeAll(func() {
+					err := FVTClientInstance.ConnectToModelServing(Insecure)
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				AfterAll(func() {
+					FVTClientInstance.DeleteIsvc(isvcName)
+					FVTClientInstance.DisconnectFromModelServing()
+				})
+
+			})
+		}
+	})
+
+	Describe("with PVC not in storage-config", Ordered, func() {
+		var isvcObject *unstructured.Unstructured
+
+		It("should fail with PVC not mounted", func() {
+			isvcObject = NewIsvcForFVT(isvcFiles[isvcWithPvcNotInStorageConfig])
+
+			obj := CreateIsvcAndWaitAndExpectFailed(isvcObject)
+
+			By("Asserting on the ISVC state")
+			ExpectIsvcFailureInfo(obj, "ModelLoadFailed", true, true, "")
+
+			FVTClientInstance.DeleteIsvc(isvcObject.GetName())
+		})
+
+		It("should load a model when allowAnyPVC", func() {
+			// This ISVC needs a new PVC which is not in the storage-config secret.
+			// The controller will update the deployment with the pvc_mount, but
+			// if the old runtime pods are still around, the ISVC will get deployed
+			// on an old runtime pod without the PVC mounted and keep failing
+			// until a new pod with the PVC is ready and the controller finally
+			// decides to move the ISVC onto the new pod that has the PVC mounted.
+			// However, this process can take a long time (how long?) so, we take
+			// some extra measures to increase our chances for quick success:
+			// - scale to 0 prohibits the new ISVC to land on an old runtime pod
+			//   that does not have the "any" PVC mounted yet
+			// - use more than 1 pod per runtime so controller will not kill new
+			//   pods that have the PVC mounted but because the ISVC is loaded on
+			//   the old pod (without the PVC) but the old pod gets kept around
+			//   instead of the new because the ISVC is still on there -- even
+			//   though its failing
+			// - allowAnyPVC needs rest-proxy enabled (not sure why)
```

Comment:
Hmm, that seems quirky. Hopefully that is something we can fix easily.

Reply:
Yes this sounds pretty odd, it should be pretty much unrelated. Do you have any more detail of the actual error/symptoms?

Reply:
Oh, this part: `allowAnyPVC needs rest-proxy enabled (not sure why)`? 

Sorry that was from an early misreading of Chin's code. I will remove it.

```go

	// collect PVCs from Predictors when the global flag 'allowAnyPVC' is enabled
	predictorPVCsMap := make(map[string]struct{})
	if cfg.AllowAnyPVC {
		restProxyEnabled := cfg.RESTProxy.Enabled

		// use a predicate function to extract the PVCs from Predictors in the registry
		f := func(p *api.Predictor) bool {
			if runtimeSupportsPredictor(rt, p, restProxyEnabled, req.Name) &&
				p.Spec.Storage != nil &&
```

Reply:
some of the other thoughts/obeservations/assumptions in my comment are no longer true since we fixed the unstable deployments

Reply:
I cleared that up -- apologies this confused rambling were my working notes while trying to figure out what was causing the deployment instabilities